atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -80,117 +65,142 @@ private CassandraTimestampBoundStore(CassandraClientPool clientPool) {
                 "Creating CassandraTimestampBoundStore object on thread {}. This should only happen once.",
                 Thread.currentThread().getName());
         this.clientPool = Preconditions.checkNotNull(clientPool, "clientPool cannot be null");
+        this.id = UUID.randomUUID();
+        DebugLogger.logger.info("The ID of this timestamp service is {}.", id);
+    }
+
+    @VisibleForTesting
+    UUID getId() {
+        return id;
     }
 
     @Override
     public synchronized long getUpperLimit() {
         DebugLogger.logger.debug("[GET] Getting upper limit");
-        return clientPool.runWithRetry(new FunctionCheckedException<Client, Long, RuntimeException>() {
-            @Override
-            public Long apply(Client client) {
-                ByteBuffer rowName = getRowName();
-                ColumnPath columnPath = new ColumnPath(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName());
-                columnPath.setColumn(getColumnName());
-                ColumnOrSuperColumn result;
-                try {
-                    result = client.get(rowName, columnPath, ConsistencyLevel.LOCAL_QUORUM);
-                } catch (NotFoundException e) {
-                    result = null;
-                } catch (Exception e) {
-                    throw Throwables.throwUncheckedException(e);
+        return clientPool.runWithRetry(client -> {
+                    ColumnPath columnPath = new ColumnPath(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName());
```

Comment:
Being introduced to `ColumnPath` here threw me - it's an entirely different level of abstraction to "how do I get the upper limit"?

It would be good to hide this by putting lines 81-85 into a separate method.